components/activity_and_insights/insights/top_channels/top_channels_line_chart/top_channels_line_chart.tsx

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React, {memo, useMemo} from 'react';
+import {useSelector} from 'react-redux';
+
+import {FormattedMessage} from 'react-intl';
+
+import {TimeFrame, TimeFrames, TopChannel, TopChannelGraphData} from '@mattermost/types/insights';
+
+import LineChart from 'components/analytics/line_chart';
+import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
+
+import './../../../activity_and_insights.scss';
+
+type Props = {
+    topChannels: TopChannel[];
+    timeFrame: TimeFrame;
+    channelLineChartData: TopChannelGraphData;
+}
+
+const TopChannelsLineChart = ({topChannels, timeFrame, channelLineChartData}: Props) => {
+    const theme = useSelector(getTheme);
+
+    const getLabels = useMemo(() => {
+        return Object.keys(channelLineChartData);
+    }, [channelLineChartData]);
+
+    const sortGraphData = useMemo(() => {
+        const labels = getLabels;
+        const values = {} as any;
+        for (let i = 0; i < labels.length; i++) {
+            const item = channelLineChartData[labels[i]];
+
+            const channelIds = Object.keys(item);
+
+            for (let j = 0; j < channelIds.length; j++) {
+                const count = item[channelIds[j]];
+                if (values[channelIds[j]]) {
+                    values[channelIds[j]].push(count);
+                } else {
+                    values[channelIds[j]] = [count];
+                }
+            }
+        }
+        return {
+            labels,
+            values,
+        };
+    }, []);
+
+    const getGraphData = useMemo(() => {
+        const data = sortGraphData;
+        const dataset = [];
+
```

Comment:
What would you think about putting the colours in an array and then doing this in a loop?

```js
const colours = [theme.buttonBg, theme.onlineIndicator, theme.awayIndicator, theme.dndIndicator, theme.newMessageSeparator];

for (i = 0; i < 5; i++) {
        if (topChannels[i]) {
            const colour = colours[i];
            dataset.push({
                fillColor: colour,
                borderColor: colour,
                pointBackgroundColor: theme.buttonBg,
                pointBorderColor: colour,
                backgroundColor: 'transparent',
                pointRadius: 0,
                hoverBackgroundColor: colour,
                label: topChannels[i].display_name,
                data: data.values[topChannels[i].id],
                hitRadius: 10,
            });
        }
}
```